### PR TITLE
Add PlayerIPParser

### PR DESCRIPTION
**Pull request by PotatoPresident**
Adds player IP log filter to alert and warn users if they post a log that contains player IPs

### CODE REVIEW

**Comment by triphora** [comment]
You might even want to delete the message and DM the message back to the sender -- I know that the suggestions do this in certain circumstances for example

**Comment by PotatoPresident** [comment]
> You might even want to delete the message and DM the message back to the sender -- I know that the suggestions do this in certain circumstances for example

Yeah, that would probably be good. Doesn't fit the current system though and didn't have the time or desire to rework it.

**Comment by gdude2002** [comment]
This doesn't take ipv6 addresses into account

File: src/main/kotlin/org/quiltmc/community/modes/quilt/extensions/logs/parsers/PlayerIPParser.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    "[a-zA-Z]+\\[/[0-9]+\\.[0-9]+\\.[0-9]+\\.[0-9]+:[0-9]+] logged in with entity id"
+
+class PlayerIPParser : BaseLogParser {
+    override suspend fun getMessages(logContent: String): List<String> {
+        val messages: MutableList<String> = mutableListOf()
+
+        if (logContent.contains(IP_REGEX)) {
+            @Suppress("MaximumLineLength")
+            messages.add(
+                "You appear to have shared player IP addresses. " +
```
Comment (gdude2002) [inline src/main/kotlin/org/quiltmc/community/modes/quilt/extensions/logs/parsers/PlayerIPParser.kt]: I'd prefer something more in line with the messages already there. For example:

`This log appears to contain players' IP addresses - please consider deleting your uploaded log, and posting it again with the IP addresses removed.`

**Review by Akarys42 — APPROVED** [review]
Works as expected, although I do have to wonder if it wouldn't be worth catching more IP addresses.

**Comment by Nicholas-Vo** [comment]
Very nice!

**Comment by NoahvdAa** [comment]
If this ends up deleting messages, could it be edited to exclude local (or just loopback) IPs? It'd be a bit annoying to have log files get deleted for a local test server with only local clients
(Although I'm not sure how often that'd happen)

**Comment by gdude2002** [comment]
> could it be edited to exclude local (or just loopback) IPs? 

Arguably it should do that in this iteration too, imo

**Comment by gdude2002** [comment]
Looks good - thanks!

**Pull request merged**